src/StringTools/WeakStringCache.Concurrent.cs

```diff
@@ -14,6 +14,7 @@ namespace Microsoft.NET.StringTools
     internal sealed partial class WeakStringCache : IDisposable
     {
         private readonly ConcurrentDictionary<int, StringWeakHandle> _stringsByHashCode;
+        private int _count;
```

Comment:
This feels can get out of sync after the `Scavenge` call (as judging by name, the goal is to remove items, while we only increment the count)

Reply:
The scavenge call is resetting this counter after it's done:
```
_scavengeThreshold = _stringsByHashCode.Count * 2;
                      _count = _stringsByHashCode.Count;
```
There might be a slight inaccuracy, however I would argue that it's fine due to the fact that we're only using it here and only as a threshold to occasionaly reduce the size of the collection.

Reply:
I have for some reason missed that part :-O
Looks good like this!